src/core/functions/table/match.cpp

```diff
@@ -948,11 +1080,16 @@ PGQMatchFunction::MatchBindReplace(ClientContext &context,
   if (ref->where_clause) {
     conditions.push_back(std::move(ref->where_clause));
   }
+
+  CheckColumnBinding(*pg_table, *ref);
```

Comment:
Having an additional validation function, instead of splitting and baking the validation logic into the multiple for loops seem to make code cleaner.